core/includes/form.inc

```diff
@@ -4907,33 +4933,51 @@ function theme_form_element($variables) {
   if (!isset($element['#title'])) {
     $element['#title_display'] = 'none';
   }
+
+  $description_before = '';
+  $description_after = '';
+  if (!empty($element['#description'])) {
+    $description = theme('form_element_description', $variables);
+
+    switch ($element['#description_display']) {
+      case 'before':
+        // It is preferable for screen readers if the help text is near/after
+        // the label of the element. So treat it the same as "before" when it
+        // comes to its placement in that case.
+        $description_before = $description;
+        break;
+
+      case 'after':
+      case 'invisible':
+        $description_after = $description;
+        break;
+    }
+  }
+
+  $label = theme('form_element_label', $variables);
+
   $prefix = isset($element['#field_prefix']) ? '<span class="field-prefix">' . $element['#field_prefix'] . '</span> ' : '';
   $suffix = isset($element['#field_suffix']) ? ' <span class="field-suffix">' . $element['#field_suffix'] . '</span>' : '';
+  $main_element = $description_before . $prefix . $element['#children'] . $suffix;
 
   switch ($element['#title_display']) {
     case 'before':
-    case 'invisible':
-      $output .= ' ' . theme('form_element_label', $variables);
-      $output .= ' ' . $prefix . $element['#children'] . $suffix . "\n";
+      $output .= ' ' . $label . ' ' . $main_element . "\n";
       break;
 
     case 'after':
-      $output .= ' ' . $prefix . $element['#children'] . $suffix;
-      $output .= ' ' . theme('form_element_label', $variables) . "\n";
+    case 'invisible':
```

Comment:
Don't agree with this. the `'invisible'` case for `'#title_display'` should be the same as `'before'` (which is the default for labels).

Reply:
@klonos this one was a mistake. I was trying to make sure `invisible` for `#description_display` was with `after` to match the approach in D8+.